components/service/glean/src/main/java/mozilla/components/service/glean/storages/TimingDistributionsStorageEngine.kt

```diff
@@ -190,8 +190,13 @@ data class TimingDistributionData(
             } catch (e: org.json.JSONException) {
                 return null
             }
-            val rawHistogramType = jsonObject.tryGetInt("histogramType") ?: return null
-            val histogramType = HistogramType.values().getOrNull(rawHistogramType) ?: return null
+            val rawHistogramType = jsonObject.tryGetString("histogram_type") ?: return null
+            var histogramType: HistogramType
+            try {
```

Comment:
nit: this could be more idiomatic by doing

`val histogramType = try {
 ...
} catch (...) {
  return null
}`

the catch here is that we should check if `return` here is returning from the scope (and assigning the value to the variable) or returning from the function. Maybe we should double-check this  anyway as I see this pattern already used for `range`, above.

Reply:
`return` always returns from the function scope.

Reply:
Got it.  I'll definitely make the syntactic simplification.  Confirmed that return returns from the function, and thus bails on the whole thing if there's a syntax error in deserialization (which I think is what we want, since this is data that we generated ourselves, so if there's a problem, it would be pretty unlikely / serious).